homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,224 @@
+
+from .const import BATTERY_ASSUMED_VOLTAGE, COORDINATOR, DOMAIN
+from .coordinator import MyPermobilCoordinator
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=50)
+
+
+@dataclass
+class PermobilRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Any], Any]
+    item: str
+
+
+@dataclass
+class PermobilSensorEntityDescription(
+    SensorEntityDescription, PermobilRequiredKeysMixin
+):
+    """Describes Permobil sensor entity."""
+
+
+SENSOR_DESCRIPTIONS: tuple[PermobilSensorEntityDescription, ...] = (
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_STATE_OF_CHARGE,
+        key="state_of_charge",
+        name="Permobil Battery Charge",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_STATE_OF_HEALTH,
+        key="state_of_health",
+        name="Permobil Battery Health",
+        icon="mdi:battery-heart-variant",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_CHARGE_TIME_LEFT,
+        key="charge_time_left",
+        name="Permobil Charge Time Left",
+        icon="mdi:battery-clock",
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_DISTANCE_LEFT,
+        key="distance_left",
+        name="Permobil Distance Left",
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_INDOOR_DRIVE_TIME,
+        key="indoor_drive_time",
+        name="Permobil Indoor Drive Time",
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x * BATTERY_ASSUMED_VOLTAGE,
+        item=BATTERY_MAX_AMPERE_HOURS,
+        key="max_watt_hours",
+        name="Permobil Battery Max Watt Hours",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
```

Comment:
A "max" is a calculation and not a long term static, and should not carry a state class.

```suggestion
```

Reply:
Changed this from `total `to `measurement`

Reply:
Or should it not have a state class at all?